node/src/components/storage.rs

```diff
@@ -477,18 +475,47 @@ impl Storage {
             metrics,
         };
 
-        if let Some(raw) =
-            component.read_state_store(&Cow::Borrowed(COMPLETED_BLOCKS_STORAGE_KEY))?
-        {
-            let (mut sequences, _) = DisjointSequences::from_vec(raw)
-                .map_err(FatalStorageError::UnexpectedDeserializationFailure)?;
+        match component.read_state_store(&Cow::Borrowed(COMPLETED_BLOCKS_STORAGE_KEY))? {
+            Some(raw) => {
+                let (mut sequences, _) = DisjointSequences::from_vec(raw)
+                    .map_err(FatalStorageError::UnexpectedDeserializationFailure)?;
 
-            // Truncate the sequences in case we removed blocks via a hard reset.
-            if let Some(&highest_block_height) = component.block_height_index.keys().last() {
-                sequences.truncate(highest_block_height);
-            }
+                // Truncate the sequences in case we removed blocks via a hard reset.
+                if let Some(&highest_block_height) = component.block_height_index.keys().last() {
+                    sequences.truncate(highest_block_height);
+                }
 
-            component.completed_blocks = sequences;
+                component.completed_blocks = sequences;
+            }
+            None => {
+                // No state so far. We can make the following observations:
+                //
+                // 1. Any block already in storage from versions prior to 1.5 (no fast-sync) MUST
+                //    have the corresponding global state in contract runtime due to the way sync
```

Comment:
```suggestion
                //    has the corresponding global state in contract runtime due to the way sync
```

(Or "blocks" above.)